bin/import-or-require.js

```diff
@@ -8,7 +8,7 @@ module.exports = function importOrRequire(file) {
     const ext = extnamePath(file);
 
     if (ext === '.mjs' || (ext === '.js' && getPackageType.sync(file) === 'module')) {
-        return import(file);
+        return import('file://' + file);
```

Comment:
I'm seeing the same issue on Windows (with node 16.9.1).

I suggest `import(pathToFileURL(file).href)`, using `const { pathToFileURL } = require('url')` which was added in node 10.12.0.

Reply:
PS. The error message that I'm getting has a more complete explanation than the message shared in the OP. Running `tape test.js` (that's a file on `C:\`) yields:

```
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default
ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'
```

Reply:
@ljharb See https://github.com/nodejs/node/commit/a0846326dd42bf4ec2f91df9cade762283567511. The error message was further tweaked in a later commit, but this should suffice to explain it.

Reply:
@vweevers that makes sense, but I still need to find a way to repro it on windows (but i don't have a windows box) or a test that would have failed on windows (even though we're not running tests on windows yet)

Reply:
Would you take a PR for a basic windows workflow on GitHub Actions (just on latest node to start), with a test? Can split those into 2 PRs if you prefer that

Reply:
Yes, but I prefer not to use the `setup-node` action, and I haven't yet adapted https://github.com/ljharb/actions/tree/main/node/install to work on windows yet.

Reply:
OK, in that case I'll only send a PR for a test (if I can get existing tests to pass on Windows).